app/src/main/java/com/android/sample/ui/activity/ActivityForm.kt

```diff
@@ -0,0 +1,400 @@
+package com.android.sample.ui.activity
+
+import android.content.Context
+import android.graphics.Bitmap
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.AccessTime
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.CalendarMonth
+import androidx.compose.material.icons.filled.HourglassTop
+import androidx.compose.material3.Button
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ExposedDropdownMenuBox
+import androidx.compose.material3.ExposedDropdownMenuDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.PopupProperties
+import com.android.sample.R
+import com.android.sample.model.activity.ActivityType
+import com.android.sample.model.activity.Category
+import com.android.sample.model.activity.types
+import com.android.sample.model.image.ImageViewModel
+import com.android.sample.model.map.Location
+import com.android.sample.model.profile.User
+import com.android.sample.model.profile.interestStringValues
+import com.android.sample.model.profile.interestsCategories
+import com.android.sample.resources.C.Tag.BUTTON_HEIGHT
+import com.android.sample.resources.C.Tag.BUTTON_WIDTH
+import com.android.sample.resources.C.Tag.LARGE_PADDING
+import com.android.sample.resources.C.Tag.SMALL_PADDING
+import com.android.sample.resources.C.Tag.STANDARD_PADDING
+import com.android.sample.resources.C.Tag.TOP_TITLE_SIZE
+import com.android.sample.ui.camera.Carousel
+import com.android.sample.ui.components.AttendantPreview
+import com.android.sample.ui.components.MyDatePicker
+import com.android.sample.ui.components.MyTimePicker
+import com.android.sample.ui.dialogs.AddUserDialog
+import com.google.firebase.Timestamp
+import com.vanpra.composematerialdialogs.MaterialDialogState
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ActivityForm(
```

Comment:
Even if the number of changes is astronomic. This is just a change of perspective in the way we fill activities, it is more compact now

Reply:
Yes, absolutly